includes/managers/controls.php

```diff
@@ -461,12 +478,42 @@ private function register_controls() {
 	 *
 	 * @since 1.0.0
 	 * @access public
+	 * @deprecated 3.5.0 Use `$this->register()` instead.
 	 *
 	 * @param string       $control_id       Control ID.
 	 * @param Base_Control $control_instance Control instance, usually the
 	 *                                       current instance.
 	 */
 	public function register_control( $control_id, Base_Control $control_instance ) {
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function(
+			__METHOD__,
+			'3.5.0',
+			'register'
+		);
+
+		$this->register( $control_instance, $control_id );
+	}
+
+	/**
+	 * Register control.
+	 *
+	 * This method adds a new control to the controls list. It adds any given
+	 * control to any given control instance.
+	 *
+	 * @since 3.5.0
+	 * @access public
+	 *
+	 * @param Base_Control $control_instance Control instance, usually the current instance.
+	 * @param string       $control_id       Control ID. Deprecated parameter.
+	 *
+	 * @return void
+	 */
+	public function register( Base_Control $control_instance, $control_id = null ) {
+		// TODO: For BC. Remove in the future.
+		if ( ! $control_id ) {
```

Comment:
If the `$control_id` is set add deprecation notice.

Reply:
Done